### PR TITLE
fix: add random credential file

### PR DESCRIPTION
**Pull request by deer-hang**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/images/obs-csi-plugin/socket-server.go [hunk 1]
```diff
@@ -148,8 +154,11 @@ func mountHandler(parameters map[string]string) error {
 		return err
 	}
 	glog.Infof("Success to mount cmd: %s", mntCmd)
-	if err := os.RemoveAll(parameters[credential]); err != nil {
-		glog.Errorf("Failed to remove passwd file, %v", err)
+	dir := filepath.Dir(parameters[credential])
+	if strings.HasPrefix(dir, credentialDir) {
+		if err := os.RemoveAll(dir); err != nil {
+			glog.Errorf("Failed to remove passwd file, %v", err)
+		}
```
Comment (chengxiangdong) [inline cluster/images/obs-csi-plugin/socket-server.go:161]: we should delete file in the defer

File: pkg/obs/nodeserver.go [hunk 2]
```diff
@@ -87,7 +88,8 @@ func (ns *nodeServer) NodePublishVolume(_ context.Context, req *csi.NodePublishV
 	if err := ns.Mount.MakeDir(targetPath); err != nil {
 		return nil, status.Errorf(codes.Internal, "Failed to make dir: %s, error: %v", targetPath, err)
 	}
-	if err := createCredentialFile(ns.Driver.cloud.Global.AccessKey, ns.Driver.cloud.Global.SecretKey); err != nil {
+	credentialFile, err := createCredentialFile(ns.Driver.cloud.Global.AccessKey, ns.Driver.cloud.Global.SecretKey)
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:91]: It is best to delete `credentialFile` file after the fun finishes

File: pkg/obs/nodeserver.go [hunk 3]
```diff
@@ -119,20 +121,21 @@ func (ns *nodeServer) NodePublishVolume(_ context.Context, req *csi.NodePublishV
 	return &csi.NodePublishVolumeResponse{}, nil
 }
 
-func createCredentialFile(accessKey, secretKey string) error {
+func createCredentialFile(accessKey, secretKey string) (string, error) {
+	credentialFile := fmt.Sprintf("%s/%s/passwd-obsfs", credentialDir, uuid.New().String())
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:125]:
```suggestion
	credentialFile := fmt.Sprintf("%s/%s", credentialDir, uuid.New().String())
```

**Review by chengxiangdong — COMMENTED** [review]
Overall good, the file should be deleted in `defer`.

File: cluster/images/obs-csi-plugin/socket-server.go (hunk 1 above)
Comment (deer-hang) [reply cluster/images/obs-csi-plugin/socket-server.go:161]: done, it has been fixed

File: pkg/obs/nodeserver.go (hunk 2 above)
Comment (deer-hang) [reply pkg/obs/nodeserver.go:91]: done. add "defer deleteCredentialFile" function in nodeserver

File: pkg/obs/nodeserver.go (hunk 3 above)
Comment (deer-hang) [reply pkg/obs/nodeserver.go:125]: done

File: pkg/obs/nodeserver.go [hunk 4]
```diff
@@ -87,10 +88,15 @@ func (ns *nodeServer) NodePublishVolume(_ context.Context, req *csi.NodePublishV
 	if err := ns.Mount.MakeDir(targetPath); err != nil {
 		return nil, status.Errorf(codes.Internal, "Failed to make dir: %s, error: %v", targetPath, err)
 	}
-	if err := createCredentialFile(ns.Driver.cloud.Global.AccessKey, ns.Driver.cloud.Global.SecretKey); err != nil {
+
+	credentialFile := fmt.Sprintf("%s/%s", credentialDir, uuid.New().String())
+	defer deleteCredentialFile(credentialFile)
```
Comment (chengxiangdong) [inline pkg/obs/nodeserver.go:93]: this file should be deleted after creation, please move this line after lint 99

Comment (deer-hang) [reply pkg/obs/nodeserver.go:93]: done

File: pkg/utils/sha256.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	for _, k := range keys {
+		element := fmt.Sprintf("%s=%s", k, valueMap[k])
+		elements = append(elements, element)
+	}
+	src := strings.Join(elements, "&")
+
+	m := sha256.New()
+	m.Write([]byte(src))
+	res := hex.EncodeToString(m.Sum(nil))
+	return res
```
Comment (chengxiangdong) [inline pkg/utils/sha256.go:27]:
```suggestion
	return res[0, 128]
```
length 256 is t0o long, cut at 128 pos

**Pull request merged**